### DISCUSSION TITLE
the complexity of environment on highway-env

### DISCUSSION THREAD

**Discussion by FivaTagin** [eleurent/highway-env#202] [Q&A] [answered]
Hi,

I am currently using rl-agent and highway environment to create my experimenting environment. 

It is working really well, thanks for your fantastic work!

However, I have gotten some problems to set the complexity on other cars on highway-env

1 . the aggressive on other cars
Currently I have seem only minor cars will change their lane. 
The behaviour does not really change when I using "other_vehicles_type": "highway_env.vehicle.behavior.AggressiveVehicle" as my env config. 
I am wondering how to increase it's aggressive to gain the complexity and challenge on the rl-algorithms. 

2. the setting of duration 
the original value of duration is 40(sec), I have make it as  2000 up to make the system as the long turn running
Although, the rl-agent can operate in a long turn duration. Other car will not be shown after a period of time. 
I do appreciate it if you can provide a hint about how to make other cars exist during the entire duration 
I have tried to make action as ContinuousAction, but it does not work. 

Much appreciated,

**Comment by eleurent** [comment, marked as answer]
Hi!

1. regarding the lane changing model, it is the [MOBIL](https://mtreiber.de/publications/MOBIL_TRB.pdf) model, which you can check for reference. If you want to increase the number of lane changes, you can change the parameters of `IDMVehicle` in the following way:
  * increase POLITENESS
  * decrease LANE_CHANGE_MIN_ACC_GAIN
  * increase LANE_CHANGE_MAX_BRAKING_IMPOSED
2. that would require some changes in the environment implementation. You can either
  * make a new environment with a circular road (with a large radius)
  * add a mechanism in env.step() that automatically spawns new vehicles in front of the ego vehicle, and deletes vehicles that are too far. You can look at the `_spawn_vehicle` method of `IntersectionEnv` for inspiration.

**Reply by FivaTagin** [reply]
it is really helpful, appreciate for your answers!
